MapboxCoreNavigationTests/DistanceFormatterTests.swift

```diff
@@ -0,0 +1,65 @@
+import XCTest
+import CoreLocation
+@testable import MapboxCoreNavigation
+
+let oneMile: CLLocationDistance = metersPerMile
+let oneFeet: CLLocationDistance = 0.3048
+
+class DistanceFormatterTests: XCTestCase {
+    
+    var spokenDistanceFormatter = SpokenDistanceFormatter(approximate: true)
+    var distanceFormatter = DistanceFormatter(approximate: true)
+    
+    override func setUp() {
+        super.setUp()
+        spokenDistanceFormatter.unitStyle = .long
+    }
+    
+    func assertDistance(_ distance: CLLocationDistance, spoken: String, displayed: String) {
+        let spokenString = spokenDistanceFormatter.string(from: distance)
+        let displayedString = distanceFormatter.string(from: distance)
+        XCTAssert(spokenString.contains(spoken), "Spoken '\(spokenString)' should be equal to '\(spoken)'")
+        XCTAssert(displayedString.contains(displayed), "Displayed: '\(displayedString)' should be equal to \(displayed)")
+    }
+    
+    func testDistanceFormatters_US() {
+        spokenDistanceFormatter.numberFormatter.locale = Locale(identifier: "en-US")
+        distanceFormatter.numberFormatter.locale = Locale(identifier: "en-US")
+        
+        assertDistance(oneFeet*50,      spoken: "50 feet",              displayed: "50 ft")
+        assertDistance(oneFeet*100,     spoken: "100 feet",             displayed: "100 ft")
+        assertDistance(oneFeet*249,     spoken: "250 feet",             displayed: "250 ft")
+        assertDistance(oneFeet*305,     spoken: "300 feet",             displayed: "300 ft")
+        assertDistance(oneFeet*997,     spoken: "1,000 feet",           displayed: "1,000 ft")
+        assertDistance(oneMile*0.25,    spoken: "a quarter mile",       displayed: "0.2 mi")
```

Comment:
If we’re using the formatter to count down to a maneuver, I suppose it’s better to undercount than to overcount.